File: Backend/src/main/java/cat/copernic/backendProjecte3/api/dto/reserva/ReservaDetallResponse.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package cat.copernic.backendProjecte3.api.dto.reserva;

import java.math.BigDecimal;
import java.time.LocalDate;

/**
 * DTO utilitzat per retornar el detall complet d'una reserva.
 *
 * Aquesta classe representa la informació que el backend envia al client quan
 * es consulta el detall d'una reserva concreta. Inclou tant les dades de la
 * reserva com la informació del vehicle associat.
 *
 * Informació inclosa: - Dades de la reserva (id, client, dates, import total i
 * fiança) - Informació del vehicle reservat (matrícula, tipus, motor, potència,
 * color i preu per hora) - Ruta o identificador de la fotografia del vehicle
 *
 * Aquest objecte és utilitzat principalment per l'endpoint de detall de reserva
 * del {@link cat.copernic.backendProjecte3.api.controller.ReservaController}.
 *
 * @author orjon
 */
public class ReservaDetallResponse {

    private Long idReserva;
    private String clientEmail;

    private LocalDate dataInici;
    private LocalDate dataFi;

    private BigDecimal importTotal;
    private BigDecimal fianca;

    // Vehicle (dades + “foto”)
    private String vehicleMatricula;
    private String vehicleTipus;
    private String vehicleMotor;
    private String vehiclePotencia;
    private String vehicleColor;
    private BigDecimal vehiclePreuHora;

    private String vehicleFoto;

    /**
     * Retorna l'identificador de la reserva.
     *
     * @return id de la reserva
     */
    public Long getIdReserva() {
        return idReserva;
    }

    /**
     * Defineix l'identificador de la reserva.
     *
     * @param idReserva identificador de la reserva
     */
    public void setIdReserva(Long idReserva) {
        this.idReserva = idReserva;
    }

    /**
     * Retorna el correu electrònic del client que ha fet la reserva.
     *
     * @return email del client
     */
    public String getClientEmail() {
        return clientEmail;
    }

    /**
     * Defineix el correu electrònic del client.
     *
     * @param clientEmail email del client
     */
    public void setClientEmail(String clientEmail) {
        this.clientEmail = clientEmail;
    }

    /**
     * Retorna la data d'inici de la reserva.
     *
     * @return data d'inici
     */
    public LocalDate getDataInici() {
        return dataInici;
    }

    /**
     * Defineix la data d'inici de la reserva.
     *
     * @param dataInici data d'inici
     */
    public void setDataInici(LocalDate dataInici) {
        this.dataInici = dataInici;
    }

    public LocalDate getDataFi() {
        return dataFi;
    }

    public void setDataFi(LocalDate dataFi) {
        this.dataFi = dataFi;
    }

    public BigDecimal getImportTotal() {
        return importTotal;
    }

    public void setImportTotal(BigDecimal importTotal) {
        this.importTotal = importTotal;
    }

    public BigDecimal getFianca() {
        return fianca;
    }

    public void setFianca(BigDecimal fianca) {
        this.fianca = fianca;
    }

    /**
     * Retorna la matrícula del vehicle reservat.
     *
     * @return matrícula del vehicle
     */
    public String getVehicleMatricula() {
        return vehicleMatricula;
    }

    /**
     * Defineix la matrícula del vehicle reservat.
     *
     * @param vehicleMatricula matrícula del vehicle
     */
    public void setVehicleMatricula(String vehicleMatricula) {
        this.vehicleMatricula = vehicleMatricula;
    }

    public String getVehicleTipus() {
        return vehicleTipus;
    }

    public void setVehicleTipus(String vehicleTipus) {
        this.vehicleTipus = vehicleTipus;
    }

    public String getVehicleMotor() {
        return vehicleMotor;
    }

    public void setVehicleMotor(String vehicleMotor) {
        this.vehicleMotor = vehicleMotor;
    }

    public String getVehiclePotencia() {
        return vehiclePotencia;
    }

    public void setVehiclePotencia(String vehiclePotencia) {
        this.vehiclePotencia = vehiclePotencia;
    }

    public String getVehicleColor() {
        return vehicleColor;
    }

    public void setVehicleColor(String vehicleColor) {
        this.vehicleColor = vehicleColor;
    }

    public BigDecimal getVehiclePreuHora() {
        return vehiclePreuHora;
    }

    public void setVehiclePreuHora(BigDecimal vehiclePreuHora) {
        this.vehiclePreuHora = vehiclePreuHora;
    }

    /**
     * Retorna la ruta o identificador de la fotografia del vehicle.
     *
     * @return ruta de la fotografia
     */
    public String getVehicleFoto() {
        return vehicleFoto;
    }

    /**
     * Defineix la ruta o identificador de la fotografia del vehicle.
     *
     * @param vehicleFoto ruta de la fotografia
     */
    public void setVehicleFoto(String vehicleFoto) {
        this.vehicleFoto = vehicleFoto;
    }
}

File: Mobile/app/src/main/java/cat/copernic/odecoches/features/reservations/data/remote/ReservaDetallResponse.kt
package cat.copernic.odecoches.features.reservations.data.remote
/**
 * Classe de dades que representa el detall complet d'una reserva retornat pel backend.
 *
 * Aquesta resposta s'utilitza quan l'usuari consulta el detall d'una reserva
 * concreta des de l'aplicació mòbil. Inclou informació tant de la reserva
 * com del vehicle associat.
 *
 * Informació de la reserva:
 * - identificador de la reserva
 * - correu electrònic del client
 * - dates d'inici i finalització
 * - import total de la reserva
 * - fiança associada
 *
 * Informació del vehicle:
 * - matrícula
 * - tipus de vehicle
 * - motor
 * - potència
 * - color
 * - preu per hora
 * - fotografia del vehicle
 *
 * @property idReserva identificador de la reserva
 * @property clientEmail correu electrònic del client que ha fet la reserva
 * @property dataInici data d'inici de la reserva
 * @property dataFi data de finalització de la reserva
 * @property importTotal import total de la reserva
 * @property fianca import de la fiança
 * @property vehicleMatricula matrícula del vehicle reservat
 * @property vehicleTipus tipus de vehicle
 * @property vehicleMotor tipus de motor del vehicle
 * @property vehiclePotencia potència del vehicle
 * @property vehicleColor color del vehicle
 * @property vehiclePreuHora preu per hora del vehicle
 * @property vehicleFoto ruta o identificador de la fotografia del vehicle
 */
data class ReservaDetallResponse(
    val idReserva: Long? = null,
    val clientEmail: String? = null,
    val dataInici: String? = null,
    val dataFi: String? = null,
    val importTotal: Double? = null,
    val fianca: Double? = null,

    val vehicleMatricula: String? = null,
    val vehicleTipus: String? = null,
    val vehicleMotor: String? = null,
    val vehiclePotencia: Int? = null,
    val vehicleColor: String? = null,
    val vehiclePreuHora: Double? = null,
    val vehicleFoto: String? = null
)